pkg/agent/util/net_windows.go

```diff
@@ -177,7 +177,7 @@ func RemoveManagementInterface(networkName string) error {
 	var err error
 	var maxRetry = 3
 	var i = 0
-	cmd := fmt.Sprintf("Get-VMSwitch -Name %s  | Set-VMSwitch -AllowManagementOS $false ", networkName)
+	cmd := fmt.Sprintf("Get-VMSwitch -ComputerName $(hostname) -Name %s  | Set-VMSwitch -ComputerName $(hostname) -AllowManagementOS $false ", networkName)
```

Comment:
Sorry, could you explain more about the issue? Doesn't ComputerName already default to local computer?
> -ComputerName
Specifies one or more Hyper-V hosts on which the virtual switch is to be configured. NetBIOS names, IP addresses, and fully qualified domain names are allowable. The default is the local computer. Use localhost or a dot (.) to specify the local computer explicitly.

Reply:
This command could fail in TKG cluster environment, and we found the different DNS name can cause this problem. Since we don't know much about Hyper-V‘s implementation，we guess it has a cache-like design to get hostname from DNS name periodically if the default name is invalid. On the contrary, using command with hostname never failed in our experiment. @tnqn